src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -165,6 +225,11 @@ public IList<JsonPropertyInfo> Properties
             {
                 VerifyMutable();
 
+                if (Kind == JsonTypeInfoKind.None)
+                {
+                    ThrowHelper.ThrowInvalidOperationException_JsonTypeInfoOperationNotPossibleForKind(Kind);
```

Comment:
You added an exception to the xml docs, but does that map to this?  It doesn't seem so.